frontend/src/pages/GraphPF/TrafficAnimation/TrafficPointRendererPF.tsx

```diff
@@ -0,0 +1,178 @@
+      translate: `${moveEndX}px ${moveEndY}px`,
+      display: isInfinite ? '' : 'none'
+    };
+    // this acts like a delay at the end, the animation continues but nothing is visible
+    if (percentVisible < 100) {
+      moveAnimation[`${percentVisible}.1%`] = { display: 'none' };
+      moveAnimation['100%'] = { display: 'none' };
+    }
+  }
+  return keyframes(moveAnimation);
+}
+
+export class TrafficPointCircleRenderer extends TrafficPointRenderer {
+  readonly animationDuration: string;
+  readonly backgroundColor: string;
+  readonly borderColor: string;
+  readonly percentVisible: number;
+  readonly radius: number;
+
+  constructor(
+    animationDuration: string,
+    percentVisible: number,
+    radius: number,
+    backgroundColor: string,
+    borderColor: string
+  ) {
+    super();
+    this.animationDuration = animationDuration;
+    this.backgroundColor = backgroundColor;
+    this.borderColor = borderColor;
+    this.percentVisible = percentVisible;
+    this.radius = radius;
+  }
+
+  private getStyle(moveAnimation: string, isInfinite: boolean): string {
+    return kialiStyle({
+      animationDuration: this.animationDuration,
+      animationFillMode: 'forwards',
+      animationIterationCount: isInfinite ? 'infinite' : 1,
+      animationName: moveAnimation,
+      animationTimingFunction: 'linear',
+      fill: this.backgroundColor,
+      stroke: this.borderColor
+    });
+  }
+
+  render(
+    edge: Edge,
+    animationDelay: string,
+    isInfinite: boolean,
+    onAnimationEnd?: React.AnimationEventHandler
+  ): React.SVGProps<SVGCircleElement> {
+    const startPoint = edge.getStartPoint();
+    const moveAnimation = getMoveAnimation(edge, this.percentVisible, isInfinite);
+
+    // use random # to ensure the key is not repeat, or it can be ignored by the render
+    const key = `point-circle-${Math.random()}`;
+    return (
+      <circle
+        id={key}
+        key={key}
+        className={this.getStyle(moveAnimation, isInfinite)}
+        style={{ animationDelay: animationDelay }}
+        cx={startPoint.x}
+        cy={startPoint.y}
+        r={`${this.radius}`}
+        onAnimationEnd={onAnimationEnd}
+      />
+    );
+  }
+}
+
+export class TrafficPointDiamondRenderer extends TrafficPointRenderer {
+  readonly animationDuration: string;
+  readonly backgroundColor: string;
+  readonly borderColor: string;
+  readonly percentVisible: number;
+  readonly radius: number;
+
+  constructor(
+    animationDuration: string,
+    percentVisible: number,
+    radius: number,
+    backgroundColor: string,
+    borderColor: string
+  ) {
+    super();
+    this.animationDuration = animationDuration;
+    this.backgroundColor = backgroundColor;
+    this.borderColor = borderColor;
+    this.percentVisible = percentVisible;
+    this.radius = radius;
+  }
+
+  private getStyle(moveAnimation: string, isInfinite: boolean): string {
+    return kialiStyle({
+      animationDuration: this.animationDuration,
```

Comment:
The diamond should remain hidden until it starts moving.
```suggestion
      opacity: 0,
      animationDuration: this.animationDuration,
```